### PR TITLE
feat: tcp-socket

### PR DESCRIPTION
**Pull request by wedancedalot**

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@ use serde::Deserialize;
 
 #[derive(Deserialize)]
 pub struct Config {
-    pub zmq_port: usize,
+    pub zmq_port: u16,
```
Comment (RequescoS) [inline src/config.rs:5]: tcp_port?

File: src/sender.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    pub fn publish(&self, message: Vec<u8>) -> Result<(), GeyserError> {
+        let mut conns_to_remove = Vec::new();
+        let mut send_errs = 0;
+
+        {
+            let conns = self.conns.read().unwrap();
+            for (i, conn) in conns.iter().enumerate() {
+                if let Err(e) = conn.try_send(message.clone()) {
```
Comment (RequescoS) [inline src/sender.rs:27]: maybe it's better to use map_err()?

File: src/sender.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        println!("TCP server listening on port {}", port);
+
+        let conns = self.conns.clone();
+
+        thread::spawn(move || {
+            for stream in listener.incoming() {
+                match stream {
+                    Ok(stream) => {
+                        let conns = conns.clone();
+                        let (tx, rx): (SyncSender<Vec<u8>>, Receiver<Vec<u8>>) =
```
Comment (RequescoS) [inline src/sender.rs:66]: unnecessary explicit indication of types?

File: src/sender.rs (hunk 2 above)
Comment (wedancedalot) [reply src/sender.rs:27]: we need to continue sending to other sockets, if one failed, not return from func

File: src/metrics.rs [hunk 4]
```diff
@@ -25,8 +31,11 @@ impl Metrics {
 impl std::fmt::Display for Metrics {
     fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
         fmt.debug_struct("geyser-zmq")
```
Comment (RequescoS) [inline src/metrics.rs:33]: err in struct name (zmq)

**Review by RequescoS — APPROVED** [review]

**Pull request merged**